src/Microsoft.Performance.SDK/Processing/FileDataSourceAttribute.cs

```diff
@@ -112,4 +120,78 @@ public override string ToString()
             return toString.ToString();
         }
     }
+
+    /// <summary>
+    ///     Attribute to mark a custom data source implementation as processing
+    ///     files that do not have extensions.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false, Inherited = false)]
+    public sealed class ExtensionlessFileDataSourceAttribute
+        : DataSourceAttribute,
+          IEquatable<ExtensionlessFileDataSourceAttribute>
+    {
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="FileDataSourceAttribute"/>
+        ///     class.
+        /// </summary>
+        public ExtensionlessFileDataSourceAttribute()
```

Comment:
for the same reason as the Directory attribute, I feel like we shouldn't have this ctor